### ISSUE TITLE
Support ts 5.5 tsconfig.json with ${configDir}

### ISSUE THREAD

**Issue by Q16solver** [timocov/dts-bundle-generator#324]
**Bug report**

Hello! Recently, ts 5.5 was announced in beta with new feature https://devblogs.microsoft.com/typescript/announcing-typescript-5-5-beta/#the-configdir-template-variable-for-configuration-files
This lets us share tsconfig across packages, and I wanted to use it for my monorepo, but dts-bundle-generator gives me error saying it cannot find the imports in my library code. I think it's because the dependency isn't correct? I'm not sure what the standard way is to solve it here, I can think of perhaps 2 ways:

- Have a "next" branch that ups the deps to the beta version as well and simply just publish it as an alpha/beta
- Have the typescript dep as a peerDep? and have userland install the required version? I'm not sure if that actually works, just a thought

**Input code**

```ts
// place your code here
```

**Expected output**

```ts
// place your code here
```

**Actual output**

```ts
// place your code here
```

**Additional context**
Add any other context about the problem here (CLI options, etc)

**Comment by timocov** [comment]
Hey @Q16solver,

> but dts-bundle-generator gives me error saying it cannot find the imports in my library code

can you share what exactly does not work? I just tried the latest version of typescript simply adding `${configDir}` in the config and it seems working fine (similar to the link you shared). If you have a minimal repro of your issue it would be much appreciated.

**Comment by Q16solver** [comment]
@timocov Yep sure! Here's the minimal repro https://stackblitz.com/edit/github-hlcarv-yvtyug?file=tsconfig.build.json,tsconfig.json,package.json,src%2Findex.ts
Please run `pnpm types:gen` for the issue.
It complains that it cannot find the imports when I have ${configDir}, but once I remove that from the tsconfig.json, it successfully outputs dist/index.d.ts as normal

**Comment by timocov** [comment]
Thanks @Q16solver!

> Have the typescript dep as a peerDep? and have userland install the required version? I'm not sure if that actually works, just a thought

I checked [other similar repos](https://github.com/JoshuaKGoldberg/ts-api-utils/blob/2d6af536e6ecf9fae9a2db445dbe1db115955afa/package.json#L97-L99) and it seems this is the way to go. But I think this would be a breaking change tho.

Would you like to create a PR?
